src/main/java/thrift/logic/parser/ListCommandParser.java

```diff
@@ -0,0 +1,57 @@
+package thrift.logic.parser;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import thrift.commons.core.Messages;
+import thrift.logic.commands.ListCommand;
+import thrift.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new ListCommand object
+ */
+public class ListCommandParser implements Parser<ListCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the ListCommand
+     * and returns a ListCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public ListCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, CliSyntax.PREFIX_MONTH ,CliSyntax.PREFIX_TAG);
+        Optional<String> monthArg = argMultimap.getValue(CliSyntax.PREFIX_MONTH);
+        Optional<String> tagArg = argMultimap.getValue(CliSyntax.PREFIX_TAG);
```

Comment:
The variables should be:
```suggestion
        Month month = ParserUtil.parseMonth(argMultimap.getValue(CliSyntax.PREFIX_MONTH).get());
        Tag tag = ParserUtil.parseTag(argMultimap.getValue(CliSyntax.PREFIX_MONTH).get());
```

You would need to implement the method that I mentioned in the previous comment!